webpack/InsightsCloudSync/Components/InsightsTable/InsightsTable.js

```diff
@@ -0,0 +1,117 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+import React, { useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { Pagination, PaginationVariant } from '@patternfly/react-core';
+import { Table, TableHeader, TableBody } from '@patternfly/react-table';
+import { useForemanSettings } from 'foremanReact/Root/Context/ForemanContext';
+import SelectAllAlert from './SelectAllAlert';
+import { columns, actions } from './InsightsTableConstants';
+import TableEmptyState from './components/EmptyState';
+import {
+  modifySelectedRows,
+  getSortColumnIndex,
+  getPerPageOptions,
+} from './InsightsTableHelpers';
+import './table.scss';
+
+const InsightsTable = ({
+  page,
+  perPage: urlPerPage,
+  status,
+  sortBy,
+  sortOrder,
+  hits,
+  query,
+  itemCount,
+  fetchInsights,
+  onTableSetPage,
+  onTablePerPageSelect,
+  onTableSort,
+  onTableSelect,
+  selectedIds,
+  showSelectAllAlert,
+  selectAll,
+  clearAllSelection,
+  error,
+}) => {
+  const { perPage: appPerPage } = useForemanSettings();
+  const perPage = urlPerPage || appPerPage;
+
+  // acts as componentDidMount
+  useEffect(() => {
+    fetchInsights({ page, perPage, query, sortBy, sortOrder });
+  }, []);
+
+  return (
+    <React.Fragment>
+      <SelectAllAlert
+        itemCount={itemCount}
+        selectedIds={selectedIds}
+        showSelectAllAlert={showSelectAllAlert}
+        selectAll={selectAll}
+        clearAllSelection={clearAllSelection}
+      />
+      <Table
+        aria-label="Recommendations Table"
+        onSelect={(_event, isSelected, rowId) =>
+          onTableSelect(_event, isSelected, rowId, hits, selectedIds)
+        }
+        canSelectAll
+        sortBy={{ index: getSortColumnIndex(sortBy), direction: sortOrder }}
+        onSort={onTableSort}
+        cells={columns}
+        rows={modifySelectedRows(hits, selectedIds)}
+        actions={actions}
+      >
+        <TableHeader />
+        <TableBody />
+      </Table>
+      <TableEmptyState status={status} error={error} />
+      <Pagination
+        itemCount={itemCount}
+        widgetId="recommendation-pagination"
+        perPage={perPage}
+        page={page}
+        variant={PaginationVariant.bottom}
+        onSetPage={onTableSetPage}
+        onPerPageSelect={onTablePerPageSelect}
+        perPageOptions={getPerPageOptions(perPage)}
```

Comment:
This adds the current option to the array instead of adding the settings & url custom options. So for example when perPage is 5 it shows up like this: 
![Screenshot from 2021-01-07 11-17-14](https://user-images.githubusercontent.com/30431079/103874517-ec72bd80-50d9-11eb-82c0-b574e65aeaaa.png)

Reply:
Thanks, solved it in commit https://github.com/theforeman/foreman_rh_cloud/pull/423/commits/0e74f4752048333972cf2b3ed651ad596bf5b4ad